calico_containers/pycalico/util.py

```diff
@@ -662,3 +662,29 @@ def verify_ports(port_list):
                 raise RangeValidationError("Port is invalid. Value must be "
                                            "from 0 to 65535 ('{0}' given)."
                                            "".format(port))
+
+def get_ipv6_link_local(interface_name):
+    """
+    Runs IP routing commands to extract the currently assigned IPv6 link-local
+    address for an interface in this namespace.
+
+    :param interface_name: Name fo the target interface.
+    :return: A string represention of the link local address, or None (if one isn't assigned).
+    Will throw exception if not interface information exists for that name.
+    """
+    # Find which link local was assigned to the ipv6 interface
+    try:
+        ip_addr_output = check_output(["ip", "-6", "addr", "show", "dev", interface_name])
+    except (CalledProcessError, OSError, AttributeError) as e:
+        _log.debug("Failed to get IPv6 address for veth: %s. Error: %s",
+                interface_name, e)
+    _log.debug("Searching for linklocal of %s in: %s", interface_name, ip_addr_output)
+
```

Comment:
If check_output above raises an exception, ip_addr_output will be uninitialized.  You need to return None in the above except branch.

might be simpler to just move the re.search into the above try branch (essentially as it was before).